src/System.Private.Xml/src/System/Xml/Serialization/Compilation.cs

```diff
@@ -244,11 +254,31 @@ internal static Assembly LoadGeneratedAssembly(Type type, string defaultNamespac
             contract = (XmlSerializerImplementation)Activator.CreateInstance(contractType);
             if (contract.CanSerialize(type))
                 return serializer;
-
+            Log(SR.Format(SR.XmlSerializerExpiredDetails, serializerName, type.FullName), EventLogEntryType.Error);
             return null;
         }
 
+        private static void Log(string message, EventLogEntryType type)
+        {
+            new EventLogPermission(PermissionState.Unrestricted).Assert();
```

Comment:
EventLogPermission is an artifact of Code Access Security which is not a supported feature. Any class name ending with Permission and a method Assert() called doesn't need to be used in .Net Core.